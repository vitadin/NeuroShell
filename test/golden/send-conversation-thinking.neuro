%% Multi-turn conversation test with thinking blocks
%% Tests progressive conversation context and thinking block evolution
%% This verifies that thinking blocks reflect growing conversation history

\set _style dark1

\echo === Send Conversation Flow with Thinking Blocks Test ===
\echo Testing multi-turn conversation with context-aware thinking blocks

%% Start a focused conversation about a specific topic
\echo Starting conversation about programming...
\send I'm learning Python programming. Can you help me understand variables?

\echo Asking follow-up question...
\send What's the difference between local and global variables in Python?

\echo Getting more specific...
\send Can you show me an example of variable scope issues?

\echo Adding complexity to the conversation...
\send How do I avoid common variable scope mistakes in my Python functions?

\echo Final question to test conversation memory...
\send Based on our discussion about Python variables and scope, what's the best practice for naming variables?

\echo Conversation summary:
\session-show

\echo Checking conversation context:
\get[1]
\echo Latest assistant response length: ${#1} characters

\get[2] 
\echo Latest user message: ${2}