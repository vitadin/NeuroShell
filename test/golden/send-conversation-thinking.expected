Setting _style = dark1
=== Send Conversation Flow with Thinking Blocks Test ===
Testing multi-turn conversation with context-aware thinking blocks
Starting conversation about programming...

  --- OpenAI reasoning trace: ---                                             
                                                                              
  Thinking about the user's message: "I'm learning Python programming. Can you
  help me understand variables?". This helps verify the message flow in tests.
  The user sent 1 messages total, and I need to provide a helpful response.   
                                                                              
  --- End thinking ---                                                        
                                                                              
  This is a mocking reply (received 1 messages, last: I'm learning Python     
  programming. Can you help me understand variables?)                         

Asking follow-up question...

  --- OpenAI reasoning trace: ---                                             
                                                                              
  Thinking about the user's message: "What's the difference between local and 
  global variables in Python?". This helps verify the message flow in tests.  
  The user sent 3 messages total, and I need to provide a helpful response.   
                                                                              
  --- End thinking ---                                                        
                                                                              
  This is a mocking reply (received 3 messages, last: What's the difference   
  between local and global variables in Python?)                              

Getting more specific...

  --- OpenAI reasoning trace: ---                                             
                                                                              
  Thinking about the user's message: "Can you show me an example of variable  
  scope issues?". This helps verify the message flow in tests. The user sent 5
  messages total, and I need to provide a helpful response.                   
                                                                              
  --- End thinking ---                                                        
                                                                              
  This is a mocking reply (received 5 messages, last: Can you show me an      
  example of variable scope issues?)                                          

Adding complexity to the conversation...

  --- OpenAI reasoning trace: ---                                             
                                                                              
  Thinking about the user's message: "How do I avoid common variable scope    
  mistakes in my Python functions?". This helps verify the message flow in    
  tests. The user sent 7 messages total, and I need to provide a helpful      
  response.                                                                   
                                                                              
  --- End thinking ---                                                        
                                                                              
  This is a mocking reply (received 7 messages, last: How do I avoid common   
  variable scope mistakes in my Python functions?)                            

Final question to test conversation memory...

  --- OpenAI reasoning trace: ---                                             
                                                                              
  Thinking about the user's message: "Based on our discussion about Python    
  variables and scope, what's the best practice for naming variables?". This  
  helps verify the message flow in tests. The user sent 9 messages total, and 
  I need to provide a helpful response.                                       
                                                                              
  --- End thinking ---                                                        
                                                                              
  This is a mocking reply (received 9 messages, last: Based on our discussion 
  about Python variables and scope, what's the best practice for naming       
  variables?)                                                                 

Conversation summary:
Session: Session 1 (ID: 00000001)
System: You are a helpful assistant.
Created: 2025-01-01 00:00:01
Updated: 2025-01-01 00:00:22
Messages: 10 total

[1] user (00:00:02): I'm learning Python programming. Can you help me understand variables?
[2] assistant (00:00:05): --- OpenAI reasoning trace: ---\n\n  Thi...an you help me understand variables?) (409 chars)
[3] user (00:00:07): What's the difference between local and global variables in Python?
[4] assistant (00:00:09): --- OpenAI reasoning trace: ---\n\n  Thi...ocal and global variables in Python?) (403 chars)
[5] user (00:00:11): Can you show me an example of variable scope issues?
[6] assistant (00:00:13): --- OpenAI reasoning trace: ---\n\n  Thi...an example of variable scope issues?) (373 chars)
[7] user (00:00:15): How do I avoid common variable scope mistakes in my Python functions?
[8] assistant (00:00:17): --- OpenAI reasoning trace: ---\n\n  Thi...ope mistakes in my Python functions?) (407 chars)
[9] user (00:00:19): Based on our discussion about Python var...e best practice for naming variables? (104 chars)
[10] assistant (00:00:21): --- OpenAI reasoning trace: ---\n\n  Thi... best practice for naming variables?) (477 chars)
Checking conversation context:
1 = --- OpenAI reasoning trace: ---                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                       
  Thinking about the user's message: "Based on our discussion about Python variables and scope, what's the best practice for naming variables?". This helps verify the message flow in tests. The user sent 9 messages total, and I need to provide a helpful response.
                                                                                                                                                                                                                                                                       
--- End thinking ---                                                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                       
This is a mocking reply (received 9 messages, last: Based on our discussion about Python variables and scope, what's the best practice for naming variables?)                                                                                                          
Latest assistant response length:  characters
2 = Based on our discussion about Python variables and scope, what's the best practice for naming variables?
Latest user message: Based on our discussion about Python variables and scope, what's the best practice for naming variables?